### PR TITLE
LOVE-517: Prometheus & Grafana & JMX Exporter

### PR DESCRIPTION
**Pull request by ikarlashov**
I've added jmx_exporter jar and config for it in order to expose metrics for Prometheus.

Also there's a fix for cluster mode in xl-deploy.conf.template.j2

### CODE REVIEW

File: templates/resources/common/jmx-exporter/jmx-exporter.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+hostPort: localhost:9011
```
Comment (hierynomus) [inline templates/resources/common/jmx-exporter/jmx-exporter.yaml]: If you expose it on `localhost`, does it still listen on every network interface? Typically you would need to set it so `0.0.0.0:9011` I think...

File: templates/resources/common/modify-wrapper-linux-conf.gawk [hunk 2]
```diff
@@ -37,3 +37,13 @@ END {
...
   printf("wrapper.java.additional.%d=-XX:MaxRAMFraction=2\n", additionals + 3);
 }
+
+# Enable JMX exporter for Prometheus
+END {
+  printf("wrapper.java.additional.%d=-Dcom.sun.management.jmxremote\n", additionals + 4);
+  printf("wrapper.java.additional.%d=-Dcom.sun.management.jmxremote.authenticate=false\n", additionals + 5);
+  printf("wrapper.java.additional.%d=-Dcom.sun.management.jmxremote.ssl=false\n", additionals + 6);
+  printf("wrapper.java.additional.%d=-Dcom.sun.management.jmxremote.port=9011\n", additionals + 7);
+  printf("wrapper.java.additional.%d=-Dcom.sun.management.jmxremote.rmi.port=9011\n", additionals + 8);
```
Comment (hierynomus) [inline templates/resources/common/modify-wrapper-linux-conf.gawk]: Do we need to open up both RMI and regular port?

File: templates/resources/common/modify-wrapper-linux-conf.gawk [hunk 3]
```diff
@@ -37,3 +37,13 @@ END {
...
+  printf("wrapper.java.additional.%d=-javaagent:lib/jmx_prometheus_javaagent.jar=9100:conf/jmx-exporter.yaml\n", additionals + 9);
```
Comment (hierynomus) [inline templates/resources/common/modify-wrapper-linux-conf.gawk]: What's the `9100` here?

File: templates/resources/common/jmx-exporter/jmx-exporter.yaml (hunk 1 above)
Comment (ikarlashov) [reply templates/resources/common/jmx-exporter/jmx-exporter.yaml]: This is the configuration for jmx-exporter to connect to hostPort of local JMX. I added some comments to this yaml.

Comment (hierynomus) [reply templates/resources/common/jmx-exporter/jmx-exporter.yaml]: 👍

File: templates/resources/common/modify-wrapper-linux-conf.gawk (hunk 2 above)
Comment (ikarlashov) [reply templates/resources/common/modify-wrapper-linux-conf.gawk]: @hierynomus yep. This are default settings. If we don't specify **jmxremote.rmi.port** , then on each startup random port will be choosen.

File: templates/resources/common/modify-wrapper-linux-conf.gawk (hunk 3 above)
Comment (ikarlashov) [reply templates/resources/common/modify-wrapper-linux-conf.gawk]: @hierynomus JMX-to-Prometheus exporter is exposed on this port. With this config It listens on 0.0.0.0:9100.

Comment (hierynomus) [reply templates/resources/common/modify-wrapper-linux-conf.gawk]: Got it!

File: templates/resources/common/modify-wrapper-linux-conf.gawk (hunk 2 above)
Comment (hierynomus) [reply templates/resources/common/modify-wrapper-linux-conf.gawk]: ok!

**Pull request merged**